core/cache/branch.go

```diff
@@ -58,6 +58,28 @@ func (b *Branch) AppConfig(appName string) settings.ItemChanges {
 	return b.details.Config[appName]
 }
 
+// Created returns a Unix timestamp indicating when this generation
+// was created.
+func (b *Branch) Created() int64 {
+	return b.details.Created
+}
+
+// CreatedBy returns user who created the branch.
+func (b *Branch) CreatedBy() string {
+	return b.details.CreatedBy
+}
+
+// Completed returns a Unix timestamp indicating when this generation
+// was committed.
+func (b *Branch) Completed() int64 {
```

Comment:
Completed branches are removed from the cache.
With the working design, branches in the cache will have the zero value for `Completed` and `CompletedBy`.

There is no harm in adding these properties, but are they needed for status?

Reply:
They are needed for 'juju commits # alias list-commits', per the spec.  Per agreement, I'll leave it in, doesn't have to be used right now.